pkg/util/status.go

```diff
@@ -38,9 +42,22 @@ func IsFailed(status apiv1.JobStatus) bool {
 	return hasCondition(status, apiv1.JobFailed)
 }
 
+func IsRunning(status apiv1.JobStatus) bool {
+	return hasCondition(status, apiv1.JobRunning)
+}
+
+func IsSuspended(status apiv1.JobStatus) bool {
+	return hasCondition(status, apiv1.JobSuspended)
+}
+
 // UpdateJobConditions adds to the jobStatus a new condition if needed, with the conditionType, reason, and message
-func UpdateJobConditions(jobStatus *apiv1.JobStatus, conditionType apiv1.JobConditionType, reason, message string) error {
-	condition := newCondition(conditionType, reason, message)
+func UpdateJobConditions(
```

Comment:
This function never returns an error, but declares it, thus polluting the code of callers. I would suggest to clean this up, can be a follow up, or a preparatory pr.

Reply:
That makes sense. I would work on this on follow-up.

Reply:
Done: https://github.com/kubeflow/training-operator/pull/1868